init.lua

```diff
@@ -30,6 +30,20 @@ local function destroy(handle)
 end
 ffi.gc(cudnn.handle, destroy)
 
+cudnn.typemap = {
+   ['torch.CudaHalfTensor']   = 'CUDNN_DATA_HALF',
+   ['torch.CudaTensor']       = 'CUDNN_DATA_FLOAT',
+   ['torch.CudaDoubleTensor'] = 'CUDNN_DATA_DOUBLE',
+}
+
+-- TODO: determine if device supports true half and use true half on it
```

Comment:
Same error on Pascal, with bigger discrepancy though : 
Completed 706 asserts in 148 tests with 1 failure and 0 errors
--------------------------------------------------------------------------------
SpatialDilatedConvolution_backward_batch
error on state (backward)
LT failed: 3.5866606235504 >= 0.01
        /usr/share/lua/5.1/cunn/test.lua:1804: in function 'v'
        /usr/share/lua/5.1/cunn/test.lua:5576: in function </usr/share/lua/5.1/cunn/test.lua:5574>
--------------------------------------------------------------------------------